cli/command/service/update.go

```diff
@@ -470,6 +488,98 @@ func anyChanged(flags *pflag.FlagSet, fields ...string) bool {
 	return false
 }
 
+func addGenericResources(flags *pflag.FlagSet, spec *swarm.TaskSpec) error {
+	if !flags.Changed(flagGenericResourcesAdd) {
+		return nil
+	}
+
+	values := flags.Lookup(flagGenericResourcesAdd).Value.(*opts.ListOpts).GetAll()
+
+	generic, err := ParseGenericResources(values)
+	if err != nil {
+		return err
+	}
+
+	if spec.Resources == nil {
+		spec.Resources = &swarm.ResourceRequirements{}
+	}
+
+	if spec.Resources.Reservations == nil {
+		spec.Resources.Reservations = &swarm.Resources{}
+	}
+
+	reservations := spec.Resources.Reservations
+	l := len(reservations.GenericResources)
+
+loop:
+	for _, toAddRes := range generic {
+		r := toAddRes.DiscreteResourceSpec
```

Comment:
I believe in other update functions we build a map of the existing options keyed by name so that we can use that map for both insert without duplicates and removing items.  Could we do that here (and in remove) as well instead of the nested for loops?

Reply:
That makes the code a lot cleaner thanks!